6-Persist-Characters

BaranselKarakas opened the pull request:
Closes https://github.com/BaranselKarakas/character-sheet-creator/issues/6

Commit:
make Character a Entity

Commit:
make Character a Entity

Commit:
add CharacterRepository

Commit:
add CharacterRepository

Commit:
little changes

Commit:
little changes

Commit:
create charRepository instance in CharService

Commit:
create charRepository instance in CharService

Commit:
wip change methods to Repository(database) methods

Commit:
wip change methods to Repository(database) methods

Commit:
change methods to Repository(database) methods

Commit:
Merge branch '6-Persist-Characters' of https://github.com/BaranselKarakas/character-sheet-creator into 6-Persist-Characters

Commit:
wip add persistence

Commit:
wip add persistence

Commit:
remove no more needed lines of code

Commit:
Merge branch '6-Persist-Characters' of https://github.com/BaranselKarakas/character-sheet-creator into 6-Persist-Characters

Commit:
add min and max values to attributes

Commit:
move methods to other class

Commit:
adjust after moving methods

Commit:
refactor

Commit:
refactor

Commit:
fix method after database connection

Commit:
refactor

Commit:
changes to umlauts

Commit:
minor changes

Commit:
add tests

Commit:
add mariadb per docker

Commit:
fix typo

BaranselKarakas commented:
funzt jetzt. Die Datenbank wird zwar gelöscht wenn ich den Container runterfahre, aber denke das passt erst mal oder?

TheGor1lla commented on pom.xml:
```diff
@@ -39,7 +39,34 @@
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.dom4j</groupId>
+			<artifactId>dom4j</artifactId>
+			<version>2.1.3</version>
+		</dependency>
```
Wofür wird diese Dependency gebraucht?

BaranselKarakas replied:
Konnte ich aus meinen commits nicht mehr nachvollziehen... Habs hier entdern: [6facac1](https://github.com/BaranselKarakas/character-sheet-creator/pull/14/commits/6facac11e28c881241f505558e38da083c13d314) und funktioniert immernoch.

TheGor1lla commented on pom.xml:
```diff
@@ -39,7 +39,34 @@
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+		<dependency>
+			<groupId>org.springframework.data</groupId>
+			<artifactId>spring-data-commons</artifactId>
+		</dependency>
```
Worin unterscheidet sich diese Dependency von spring-boot-starter-data-jpa ? Brauchen wir beide?

BaranselKarakas replied:
Wenn ichs richtig verstehe, ist spring-data-commons in spring-boot-starter-data-jpa enthalten, lul. Hab das hier entfernt: [82a20eb](https://github.com/BaranselKarakas/character-sheet-creator/pull/14/commits/82a20eb197b1b771b5d1a9b9c0368131bb7904b5)

TheGor1lla commented on src/main/java/com/baran/charactersheetcreator/controller/AdminController.java:
```diff
@@ -2,45 +2,50 @@
 
+    private final CharService charService;
+
     @Autowired
```
Wenn es nur einen Constructor gibt, brauchen wir die Autowired Annotation nicht. Same für den anderen Controller

BaranselKarakas replied:
"Ab Spring 4.3 wird die @Autowired-Annotation implizit auf Konstruktoren angewendet, wenn es nur einen Konstruktor gibt."
Lul, das ist ja cool und gut zu wissen :D
Hab das hier entfernt: [397a30f](https://github.com/BaranselKarakas/character-sheet-creator/pull/14/commits/397a30f9aa2e5686fbbde78e3886af7ca4179dca)

TheGor1lla commented on src/main/java/com/baran/charactersheetcreator/controller/AdminController.java:
```diff
@@ -2,45 +2,50 @@
-        if (!getCharList().isEmpty()) {
-            model.addAttribute("myCharacterArrayList", getCharList());
+    public String showAllCharacters(Model model) {
+        if (IterableUtils.isEmpty(charService.getAllChars())) {
```
Bisschen Rattenschwanz, aber die vorherige Lösung war sauberer. Dazu müssen wir aus dem Repository allerdings wirklich eine Liste zurückgeben und kein Iterable.

BaranselKarakas replied:
Habe ich gefixt: [81eb74c](https://github.com/BaranselKarakas/character-sheet-creator/pull/14/commits/81eb74ccf0a151aca149240b69384e65a4b61722)

TheGor1lla commented on src/main/resources/templates/characterlist.html:
```diff
@@ -7,8 +7,8 @@
-<div th:each="character : ${myCharacterArrayList}">
-    <a th:href="@{'/characters/' + ${character.id}}"><span th:text="#{list.character} + ${character.name}"></span></a>
+<div th:each="character : ${allCharacter}">
+    <a th:href="@{'/characters/' + ${character.id}}"><span th:text="'Character ' + ${character.name}"></span></a>
```
Hier bietet sich i18n an

TheGor1lla commented on src/test/java/com/baran/charactersheetcreator/service/CharServiceTest.java:
```diff
@@ -0,0 +1,41 @@
+package com.baran.charactersheetcreator.service;
```
COol, tests!

TheGor1lla commented on src/main/java/com/baran/charactersheetcreator/service/CharService.java:
```diff
@@ -1,37 +1,35 @@
+    }
+
+    public Character getCharById(Integer id) {
+        return charRepository.findById(id).orElse(null);
```
Hier arbeiten wir mit null, in den Tests mit Optionals

TheGor1lla commented on src/main/java/com/baran/charactersheetcreator/controller/WebController.java:
```diff
@@ -30,22 +28,23 @@ public String getCharacterCreator(Model model) {
     public String showAllCharacters(Model model) {
-        if (!getCharList().isEmpty()) {
-            model.addAttribute("myCharacterArrayList", getCharList());
+        if (IterableUtils.isEmpty(charService.getAllChars())) {
```
Siehe List vs Iterable

BaranselKarakas replied:
Habe ich gefixt: [81eb74c](https://github.com/BaranselKarakas/character-sheet-creator/pull/14/commits/81eb74ccf0a151aca149240b69384e65a4b61722)

Commit:
remove not needed dependency

Commit:
remove unnecessary dependency

Commit:
remove unnecessary @Autowired Annotation

"Ab Spring 4.3 wird die @Autowired-Annotation implizit auf Konstruktoren
angewendet, wenn es nur einen Konstruktor gibt."

Commit:
change Iterable to List

for better  acces on the findAll() method
